--- ./acinclude.m4
+++ ./acinclude.m4
@@ -169,6 +169,8 @@ AC_CACHE_CHECK([for $1 directory], tor_cv_library_$1_dir, [
     LIBS="$tor_saved_LIBS $3"
     CPPFLAGS="$tor_saved_CPPFLAGS"
 
+    AC_MSG_WARN([Razorback razorback razorback... $tor_trydir... $5, $6])
+
     if test -z "$tor_trydir" ; then
       continue;
     fi
@@ -188,6 +190,7 @@ AC_CACHE_CHECK([for $1 directory], tor_cv_library_$1_dir, [
     # the binary?
     AC_LINK_IFELSE([AC_LANG_PROGRAM([$5], [$6])],
                    [linkable=yes], [linkable=no])
+    AC_MSG_WARN([\n$tor_trydir...\n $5...\n $6...\n $linkable...\n $tor_saved_LDFLAGS...\n $tor_saved_LIBS $3...\n $tor_saved_CPPFLAGS...\n])
 
     if test "$linkable" = yes; then
       tor_$1_any_linkable=yes     